crates/polars-arrow/src/compute/take/fixed_size_list.rs

```diff
@@ -214,10 +222,32 @@ pub(super) unsafe fn take_unchecked<O: Index>(
             }
             Some(new_validity.freeze())
         };
-        assert_eq!(count * bytes_per_element, total_bytes);
 
+        assert_eq!(count * bytes_per_element, total_bytes);
         buf.set_len(total_bytes);
 
+        let outer_validity = combine_validities_and(
+            outer_validity.as_ref(),
+            values
+                .validity()
+                // We need at least 1 element as we do `get_bit(i.unwrap_or(0))`
+                .filter(|x| !x.is_empty())
+                .map(|x| {
+                    if indices.has_nulls() {
+                        indices
+                            .iter()
+                            .map(|i| x.get_bit_unchecked(i.unwrap_or(&O::zero()).to_usize()))
```

Comment:
I don't think it should take  the `0` offset if `None`. I think it should return `false` if `None`.

Reply:
Yes, we shouldn't - this was a very silly thing to do 😝